### PR TITLE
Use different environment variable for instance setup

### LINKED ISSUE #5: Issue while cleaning up the new instance
We are using this action to setup feature instances when a developer creates pull request. We have another GitHub action that is supposed to delete the feature instance when the pull request is merged.

```yaml
name: Do cleanup

on:
  pull_request:
    types: [ closed ]

jobs:
  merge_job:
    # this job will only run if the PR has been merged
    if: github.event.pull_request.merged == true
    runs-on: ubuntu-latest
    container: axelerant/platformsh-cli:latest
    env:
      ENV_NAME: platform
    steps:
      - name: Check out repository code
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Terminate instance
        run: |
          platform project:set-remote ${PLATFORM_PROJECT_ID}
          platform environment:deactivate ${GITHUB_REF_NAME} --delete-branch
        env:
          PLATFORMSH_CLI_TOKEN: ${{ secrets.PLATFORMSH_CLI_TOKEN }}
```

`GITHUB_REF_NAME` is the branch name to which the pull request is merged, not the pull request's branch name. Due to this it fails to delete the feature instance.

### PR DESCRIPTION
**Pull request by subhojit-axl**
Fixes #5

### CODE REVIEW

**Comment by subhojit-axl** [comment]
@hussainweb @skippednote Can you please take a look.

**Comment by hussainweb** [comment]
The problem is that `GITHUB_HEAD_REF` is only used when the job runs in a pull_request workflow run ([see the docs](https://docs.github.com/en/actions/learn-github-actions/environment-variables)). If we use this in the action, then the action may only be used in such jobs and we don't want that.

We should explore making this an input parameter. That way, the calling job can set the branch to whatever they want. By default, we could use `GITHUB_REF_NAME`.

**Comment by subhojit-axl** [comment]
@hussainweb Agree. Makes sense. I will take a look.

File: action.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ inputs:
   force-push:
     description: 'Force push to Platform.sh'
     required: false
+  instance-name:
```
Comment (hussainweb) [inline action.yml:20]: How about calling this `environment-name` instead? We'll be using the same vocab as platform.sh then.

File: action.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ inputs:
...
+    description: 'Name of the platform.sh instance'
+    default: ${GITHUB_REF_NAME}
```
Comment (hussainweb) [inline action.yml:22]: Shouldn't this be `${{ ... }}`? Also, do we know that this substitution works in this section of the yml file?

Comment (subhojit-axl) [reply action.yml:22]: Yes. I am going to try this branch in one of my projects.

File: action.yml (hunk 1 above)
Comment (subhojit-axl) [reply action.yml]: Updated.

File: action.yml (hunk 2 above)
Comment (subhojit-axl) [reply action.yml]: Well, none of them are working. They are just printing the exact text `${GITHUB_REF_NAME}` if you don't pass the variable.
The documentation doesn't explicitly mentions that you can use expression, but you can use plain string.
I am trying to set the default using output parameters.

Comment (subhojit-axl) [reply action.yml]: It is now working with step output. However, it is not considering the environment name passed in the ci yaml file. I will take a look.

Comment (subhojit-axl) [reply action.yml]: The way I was passing the environment name was incorrect. That is why it was not getting considered.

**Comment by subhojit-axl** [comment]
@hussainweb This can be reviewed now.

File: deploy.sh [hunk 3]
```diff
@@ -11,7 +11,7 @@ mkdir -p ~/.ssh && chmod 0700 ~/.ssh
 cat ${GITHUB_ACTION_PATH}/known_hosts >> ~/.ssh/known_hosts
 
 platform project:set-remote ${PLATFORM_PROJECT_ID}
-PLATFORM_OPTS="-vv --activate --target ${GITHUB_REF_NAME}"
+PLATFORM_OPTS="-vv --activate --target $ENVIRONMENT_NAME"
```
Comment (hussainweb) [inline deploy.sh:14]:
```suggestion
[[ "$ENVIRONMENT_NAME" == "" ]] && ENVIRONMENT_NAME="$GITHUB_REF_NAME"
PLATFORM_OPTS="-vv --activate --target $ENVIRONMENT_NAME"
```

File: action.yml [hunk 4]
```diff
@@ -29,13 +32,22 @@ runs:
...
       with:
         token: ${{ inputs.cli-token }}
+    -
+      name: Set environment name
+      id: set_environment_name
+      shell: bash
+      run: |
+        ENVIRONMENT_NAME=${{ inputs.environment-name }}
+        [[ "$ENVIRONMENT_NAME" == "" ]] && ENVIRONMENT_NAME="$GITHUB_REF_NAME"
+        echo "::set-output name=url::$ENVIRONMENT_NAME"
```
Comment (hussainweb) [inline action.yml:42]: I am not very fond of this. I think we can handle this in the shell script itself quite robustly. See my other two suggestions.

File: action.yml [hunk 5]
```diff
@@ -29,13 +32,22 @@ runs:
...
     - run: ${{ github.action_path }}/deploy.sh
       shell: bash
       env:
         SSH_PRIVATE_KEY: ${{ inputs.ssh-private-key }}
         PLATFORM_PROJECT_ID: ${{ inputs.project-id }}
         PLATFORMSH_CLI_TOKEN: ${{ inputs.cli-token }}
         FORCE_PUSH: ${{ inputs.force-push }}
+        ENVIRONMENT_NAME: ${{ steps.set_environment_name.outputs.url }}
```
Comment (hussainweb) [inline action.yml:50]:
```suggestion
        ENVIRONMENT_NAME: ${{ inputs.environment-name }}
```

**Review by hussainweb — CHANGES_REQUESTED** [review]

File: action.yml (hunk 4 above)
Comment (hussainweb) [inline action.yml:42]:
```suggestion
```

**Pull request closed without merging**